src/views/tutorial-view/components/image-config/get-validated-image-child.tsx

```diff
@@ -0,0 +1,75 @@
+import { Children, isValidElement, ReactElement, ReactNode } from 'react'
+
+/**
+ * Given the incoming children passed to `<ImageConfig />`,
+ * validate that the children are in the expected structure.
+ *
+ * Return the child `<img />` element if the structure is expected,
+ * throw an error otherwise.
+ */
+function getValidatedImgChild(children: ReactNode): ReactElement {
```

Comment:
I've tried to make this validation function robust to avoid misuse. The idea here is to ensure we have a single `<img />` child being used with `ImageConfig`, and return that child element. This is not as straightforward as it could be since in MDX we expect the structure `<p><img /></p>`, but we could also see a single `<img />` child if authors use HTML rather than the usual `![Alt text](/img.jpg)` markdown syntax.

I spent a bit more time on this since I think it'll come in handy for MDX custom components generally. Other components that have similar validation needs: `CodeBlockConfig`, `CodeTabs`, `Tabs`.

Reply:
In looking at the tutorials repo, it seems like _all_ image references in tutorials use the plain markdown syntax. I still think this is a good thing to support though!